signer/signer.go

```diff
@@ -51,7 +36,7 @@ type UnsignedAttributes struct {
 type SignRequest struct {
 	Payload             []byte
 	PayloadContentType  string
-	CertificateChain    []*x509.Certificate
+	SignatureAlgorithm  SignatureAlgorithm
```

Comment:
Can we avoid the caller having to calculate and set Signature alg, instead derive it from SignatureProvider interface?